Helper/Cart.php

```diff
@@ -1747,4 +1756,33 @@ public function hasProductRestrictions($quote = null)
         // no restrictions
         return false;
     }
+
+    /**
+     * Create cart by item [reference,quantity]
+     * TODO: add support for logged in users
+     * TODO: add support for foreign currencies
+     * TODO: add support for multistore
+     *
+     * @param arrat $item
+     *
+     * @return array cart_data in bolt format
+     * @throws \Exception
+     */
+    public function createCartByItem($item)
+    {
+        $quoteId = $this->quoteManagement->createEmptyCart();
+        $quote = $this->quoteFactory->create()->load($quoteId);
+
+        //add item in quote
+        $product = $this->productRepository->getbyId($item['reference']);
+        $quote->addProduct($product, $item['quantity']);
+
+        $quote->reserveOrderId();
+        $quote->collectTotals()->save();
+
+        $cart_data = $this->getCartData(false,'',$quote);
```

Comment:
Something strange might happen here https://github.com/BoltApp/bolt-magento2/blob/master/Helper/Cart.php#L1133 and beyond in the getCartData method. I assume your code works but it's not clear how and why and what exactly happens in the aforementioned method. Perhaps you should add a check / use a immutableQuote quoteId instead of boltParentQuoteId when boltParentQuoteId is not set in the above line.

Reply:
@danac-gs `$quote` we created is immutable. We created it and nobody will change it. It's a reason while I pass `$quote` as the third parameter into getCartData. 

getCartData is already developed to work with the case, we can see in comments that it's for "i.e. discount code validation, bugsnag report generation".

We try to find the parent quote for $immutableQuote, but it doesn't exist so $quote is false.
Again, it's a normal situation for function, it designed to handle it.
Actually it just means we don't try to get address from parentQuote in this line
https://github.com/BoltApp/bolt-magento2/blob/0121c9ef21bb6e2eba03346288fd11c2f000f041/Helper/Cart.php#L1259

So it seems like it's enough to add a couple comments into getCartData to explain how it handles **cart.create** case. What do you think?

Reply:
I looked again and you're right. This is OK.